android/src/org/ppsspp/ppsspp/TextRenderer.java

```diff
@@ -0,0 +1,47 @@
+package org.ppsspp.ppsspp;
+import android.graphics.*;
+
+import java.nio.ByteBuffer;
+
+public class TextRenderer {
+	private static Paint p;
+	private static Paint bg;
+	static {
+		p = new Paint(Paint.SUBPIXEL_TEXT_FLAG | Paint.ANTI_ALIAS_FLAG);
+		p.setColor(Color.WHITE);
+		bg = new Paint();
+		bg.setColor(Color.BLACK);
+	}
+	public static int measureText(String string, double textSize) {
+		Rect bound = new Rect();
+		p.setTextSize((float)textSize);
+		p.getTextBounds(string, 0, string.length(), bound);
+		int w = bound.width();
+		int h = (int)(p.descent() - p.ascent() + 2.0f);
+		// Round width up to even already here to avoid annoyances from odd-width 16-bit textures which
+		// OpenGL does not like - each line must be 4-byte aligned
+		w = (w + 3) & ~1;
+		return (w << 16) | h;
+	}
+	public static int[] renderText(String string, double textSize) {
+		Rect bound = new Rect();
+		p.setTextSize((float)textSize);
+		p.getTextBounds(string, 0, string.length(), bound);
+		int w = bound.width();
+		int h = (int)(p.descent() - p.ascent() + 2.0f);
+		// Round width up to even already here to avoid annoyances from odd-width 16-bit textures which
+		// OpenGL does not like - each line must be 4-byte aligned
+		w = (w + 3) & ~1;
```

Comment:
Yeah it does, was just lazy because of Java's annoying lack of convenient ways to return 2 values. Fixing (by using a Point object).